src/scripts/main.js

```diff
@@ -1,3 +1,46 @@
 'use strict';
 
-// write code here
+const head = document.querySelector('thead');
+
+// console.log(head);
+
+head.addEventListener('click', (e) => {
+  if (e.target.tagName === 'TH') {
+    // head
+    const cells = head.querySelectorAll('th');
+    const cellsArr = [...cells];
+    const index = cellsArr.indexOf(e.target);
+
+    // body find column
+    const body = document.querySelector('tbody');
+    const bodyRows = body.querySelectorAll('tr');
+    const bodyRowsArr = [...bodyRows];
+
+    // body sort column
+    const rowsWithValues = bodyRowsArr.map((row) => {
+      const cellValue = row.children[index].textContent.trim();
+      const numericValue = parseFloat(
+        cellValue.replace('$', '').replace(',', ''),
+      );
```

Comment:
The parsing of numeric values assumes that all numeric values are formatted with a dollar sign and commas. If there are numeric values without these characters, or if there are other currency symbols, this parsing will fail. Consider adding additional checks or parsing logic to handle different numeric formats.